tests/test_lane.py

```diff
@@ -296,986 +293,1083 @@ def facetify(collection=Facets.COLLECTION_FULL,
+        # tries to build a collection based on items that are featured
+        # _within the CustomLists_.
+        lane.list_datasource = DataSource.lookup(
+            self._db, DataSource.GUTENBERG
         )
-        w, mw = _assert_expectations(
-            all_best_sellers, 2, 
-            lambda x: x.sort_title in (
-                self.fiction.sort_title, self.nonfiction.sort_title
+        self._db.commit()
+        eq_(True, lane.uses_customlists)
+        additional = [(Facets.COLLECTION_FULL, Facets.AVAILABLE_NOW, True),
+                      (Facets.COLLECTION_FULL, Facets.AVAILABLE_ALL, True)]
+        eq_(additional + default_facets, 
+            list(lane.featured_collection_facets()))
+
+    def test_apply_custom_filters(self):
+
+        # Create some works that will or won't show up in various
+        # lanes.
+        childrens_fiction = self._work(
+            fiction=True, with_license_pool=True, 
+            audience=Classifier.AUDIENCE_CHILDREN
+        )
+        nonfiction = self._work(fiction=False, with_license_pool=True)
+        childrens_fiction.target_age = tuple_to_numericrange((8,8))
+        self.add_to_materialized_view([childrens_fiction, nonfiction])
+
+        def match_works(lane, works, featured=False):
+            """Verify that calling apply_custom_filters to the given
+            lane yields the given list of works.
+            """
+            from model import MaterializedWork
+            base_query = self._db.query(MaterializedWork).join(
+                LicensePool, MaterializedWork.license_pool_id==LicensePool.id
             )
-        )
-
-        # Combine list membership with another criteria (nonfiction)
-        all_nonfiction_best_sellers = Lane(
-            self._db, self._default_library, full_name="All Nonfiction Best Sellers",
-            fiction=False,
-            list_data_source=best_seller_list_1.data_source.name
-        )
-        w, mw = _assert_expectations(
-            all_nonfiction_best_sellers, 1, 
-            lambda x: x.sort_title==self.nonfiction.sort_title
-        )
+            query, distinct = lane.apply_custom_filters(
+                self._db, base_query, MaterializedWork, featured
+            )
+            results = query.all()
+            works = sorted([(x.id, x.sort_title) for x in works])
+            materialized_works = sorted(
+                [(x.works_id, x.sort_title) for x in results]
+            )
+            eq_(works, materialized_works)
+            return distinct
+
+        # A lane may show only titles that come from a specific license source.
+        gutenberg_only = self._lane()
+        gutenberg_only.license_datasource = DataSource.lookup(
+            self._db, DataSource.GUTENBERG
+        )
+
+        distinct = match_works(gutenberg_only, [nonfiction])
+        # No custom list is involved, so there's no need to make the query
+        # distinct.
+        eq_(False, distinct)
+
+        # A lane may
```

Comment:
Looks like you forgot to check the age range filter for this lane...

Reply:
I'm not sure what this is referring to. There's no age range or audience filter in `fiction_lane`, `nonfiction_lane`, or `both_lane`, so they all find both children's and adult books.

Reply:
This comment ended up on the wrong line :(

I'll try to find it again.